File: Part 5 - Association Rule Learning/Apriori/Apriori.R
#Data Preprocessing
dataset=read.csv("Market_Basket_Optimisation.csv",header = FALSE)

#install.packages('arules')
library(arules)
dataset=read.transactions("Market_Basket_Optimisation.csv",sep = ",",
                          rm.duplicates = TRUE)
summary(dataset)
itemFrequencyPlot(dataset,topN=10)

#train the model
rules=apriori(data = dataset,parameter = list(support=0.005, confidence=0.2))

#visualize
inspect(sort(rules,by='lift')[0:10])

File: Part 5 - Association Rule Learning/Apriori/Apriori.py
#import lib

import pandas as pd
import numpy as ny
import matplotlib.pyplot as plt

#importing dataset
dataset=pd.read_csv("E:\\MY_GOAL\\Machine Learning\\Part 5 - Association Rule Learning\\Section 28 - Apriori\\Market_Basket_Optimisation.csv",header=None)
trans=[]
for i in range(0,7501):
       trans.append([str(dataset.values[i,j])for j in range(0,20)])
       
#train the model
from apyori import apriori 
rules=apriori(trans,min_support=0.005,min_confidence=0.2,min_lift=3,min_length=2)

#visualize
res=list(rules)